### PR TITLE
TYP: Added type hint to Index.repeat

### PR DESCRIPTION
**Pull request by mecopur**
- [ ] Fixes a bullet point on https://github.com/noatamir/pyladies-aug23-sprint/issues/5#issue-1805063243
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

@noatamir @phofl

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1205,7 +1206,7 @@ def _maybe_disallow_fill(self, allow_fill: bool, fill_value, indices) -> bool:
         """
 
     @Appender(_index_shared_docs["repeat"] % _index_doc_kwargs)
-    def repeat(self, repeats, axis=None):
+    def repeat(self, repeats, axis: AxisInt | None = None):
```
Comment (twoertwein) [inline pandas/core/indexes/base.py:1209]: I would prefer to change it to `axis: None = None` as the [docs](https://pandas.pydata.org/docs/reference/api/pandas.Index.repeat.html) say:

> Must be None. Has no effect but is accepted for compatibility with numpy.

Comment (twoertwein) [reply pandas/core/indexes/base.py:1209]: Same applies to the other two changes

Comment (twoertwein) [reply pandas/core/indexes/base.py:1209]: The doc-string in IndexOpsMixin.argmin/max also says that it is only there for compatibility. So I would be tempted to change even  IndexOpsMixin.argmin/max to `axis: None = None`. One could also make the argument that the type should be `int | None` as both are valid :) Let's make the change for at least repeat to be in sync with the doc-string but the other cases are up to you.

@mroeschke `IndexOpsMixin.argmin` "uses" axis exactly once - `nv.validate_minmax_axis(axis)`. Testing an unused argument seems not very helpful but removing it probably changes the runtime behavior when people expect to get an error.

Comment (mroeschke) [reply pandas/core/indexes/base.py:1209]: Yeah ideally this should still error the same if a non `None` axis was passed

**Review by twoertwein — APPROVED** [review]
You can still keep the other changes, no need to remove them :)

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @mecopur

**Pull request merged**